modules/evm/src/runner/native.rs

```diff
@@ -132,32 +151,33 @@ impl<T: Trait> RunnerT<T> for Runner<T> {
 		value: U256,
 		gas_limit: u32,
 	) -> Result<CreateInfo, Self::Error> {
-		let account_id = T::AddressMapping::into_account_id(source);
-		frame_system::Module::<T>::inc_account_nonce(&account_id);
+		debug::debug!(
+			target: "evm",
+			"create2: source {:?}, gas_limit: {:?}",
+			source,
+			gas_limit,
+		);
+
+		let vicinity = Vicinity {
+			gas_price: U256::one(),
+			origin: source,
+		};
 
-		frame_support::storage::with_transaction(|| {
-			let vicinity = Vicinity {
-				gas_price: U256::one(),
-				origin: source,
-			};
+		let config = T::config();
 
-			let config = T::config();
+		let mut substate =
+			Handler::<T>::new_with_precompile(&vicinity, gas_limit as usize, false, config, T::Precompiles::execute);
 
-			let mut substate = Handler::<T>::new_with_precompile(
-				&vicinity,
-				gas_limit as usize,
-				false,
-				config,
-				T::Precompiles::execute,
-			);
+		let code_hash = H256::from_slice(Keccak256::digest(&init).as_slice());
+		let address = substate.create_address(CreateScheme::Create2 {
+			caller: source,
+			code_hash,
+			salt,
+		});
 
-			let code_hash = H256::from_slice(Keccak256::digest(&init).as_slice());
-			let address = substate.create_address(CreateScheme::Create2 {
-				caller: source,
-				code_hash,
-				salt,
-			});
+		substate.inc_nonce(address);
```

Comment:
```suggestion
		substate.inc_nonce(source);
```

Reply:
this is intended per eip 161

Reply:
that's called here https://github.com/AcalaNetwork/Acala/pull/521/files/e0300549204ad9189127eb79c3145f5342bae0db#diff-9a78924138ebd2942f83151d282cf9216b5e8986eedd9a0051d79baa11b64409R202
I don't see caller nonce to be incremented

Reply:
will fix on #528